### PR TITLE
working on 09

### PR DESCRIPTION
**Pull request by goelvatsal**

### CODE REVIEW

File: section1/set6/exercises/06-odd-even/main.go [hunk 1]
```diff
@@ -37,4 +43,26 @@ package main
 // ---------------------------------------------------------
 
 func main() {
+	// declare variables here
+	var n int
+	var err error
+	a := os.Args
+
+	// check if os.Args is empty
+	if len(a) != 2 || len(a[1]) <= 0 {
```
Comment (goel4607) [inline section1/set6/exercises/06-odd-even/main.go]: 2nd part of OR is not required

File: section1/set6/exercises/06-odd-even/main.go [hunk 2]
```diff
@@ -37,4 +43,26 @@ package main
...
+		fmt.Println("Pick a number.")
+		return
+		// check if n is a number, not words or letters
+	} else if n, err = strconv.Atoi(a[1]); err != nil {
+		fmt.Printf("%q is not a number.\n", a[1])
+		// check if dividend is even
+	} else if n%2 == 0 && n%8 != 0 {
+		fmt.Printf("%d is an even number.\n", n)
+		// check if dividend is odd
+	} else if n&2 != 0 {
```
Comment (goel4607) [inline section1/set6/exercises/06-odd-even/main.go]: it should be a MOD operator

File: section1/set6/exercises/07-leap-year/main.go [hunk 3]
```diff
@@ -37,4 +43,26 @@ package main
 // ---------------------------------------------------------
 
 func main() {
+	// declare variables
+	a := os.Args
+	var n int
```
Comment (goel4607) [inline section1/set6/exercises/07-leap-year/main.go]: variable should be declared as close to the place where it is used

File: section1/set6/exercises/09-days-in-month/main.go [hunk 4]
```diff
@@ -88,4 +93,16 @@ package main
...
+	a := os.Args
+
+	// check if os.Args is empty
+	if len(a) != 2 {
+		fmt.Println("Give me a month name.")
+	}
+	// check correct month names entered
+	if a[1] != "january", "february", "march", "april", "may", "june",
+		"july", "august", "september", "october", "november", "december" {
+		fmt.Println("")
```
Comment (goel4607) [inline section1/set6/exercises/09-days-in-month/main.go:106]: incorrect print
Also, check for uppercase or mixed case chars
Also account for leap year

**Review by goel4607 — CHANGES_REQUESTED** [review]

File: section1/set6/exercises/06-odd-even/main.go (hunk 1 above)
Comment (goelvatsal) [reply section1/set6/exercises/06-odd-even/main.go]: fixed

File: section1/set6/exercises/06-odd-even/main.go (hunk 2 above)
Comment (goelvatsal) [reply section1/set6/exercises/06-odd-even/main.go]: fixed

File: section1/set6/exercises/07-leap-year/main.go (hunk 3 above)
Comment (goelvatsal) [reply section1/set6/exercises/07-leap-year/main.go]: fixed

File: section1/set6/exercises/09-days-in-month/main.go (hunk 4 above)
Comment (goelvatsal) [reply section1/set6/exercises/09-days-in-month/main.go:106]: currently working on 09, not completely finished
will postpone working on this tomorrow

**Review by goel4607 — APPROVED** [review]
please rem to fix the last change

**Pull request merged**